src/lib.rs

```diff
@@ -309,3 +309,106 @@ where
 
     Ok(recursive_snark)
 }
+
+#[cfg(not(target_family = "wasm"))]
+pub fn add_step<G1, G2>(
```

Comment:
The current `create_recursive_circuit` API allows for multiple steps in one call. That’s partly because the cost of witness generation can be reduced by batching. Thoughts on making this API similar?

Maybe rename to `continue_recursive_circuit` and supporting list of private inputs. It may also be possible to refactor the existing code a bit then and prevent code dupe, but I can do that clean up later as well seperately. :)